plasmapy/diagnostics/thomson.py

```diff
@@ -318,4 +393,254 @@ def spectral_density(
     # Recast as real: imaginary part is already zero
     Skw = np.real(np.sum(econtr, axis=0) + np.sum(icontr, axis=0))
 
+    # Apply an insturment function if one is provided
+    if inst_fcn is not None and callable(inst_fcn):
+        # Create an array of wavelengths of the same size as wavelengths
+        # but centered on zero
+        wspan = (np.max(wavelengths) - np.min(wavelengths)) / 2
+        eval_w = np.linspace(-wspan, wspan, num=wavelengths.size)
+        # Evaluate the insturment function
+        inst_fcn_arr = inst_fcn(eval_w)
+        # Ensure inst_fcn is normalized
+        inst_fcn_arr *= 1 / np.sum(inst_fcn_arr)
```

Comment:
```suggestion
        inst_fcn_arr = inst_fcn(eval_w)
        inst_fcn_arr *= 1 / np.sum(inst_fcn_arr)
```

This was something very difficult for myself to learn, being overly robust with comments can be a detriment.  They can easily get out of date, which often happens, and then they cause confusion.  I good rule of thumb, let the code speak for itself.  If the intent of the code is not easily understood by reading it, then add comments to clarify.  For example, `inst = inst / np.norm(inst)` is straight forwardly read as you're normalizing `inst` so you do not need a comment to clarify that.  Read over you comments and eliminate those that are just rephrasing what the code is reading as.

Reply:
Point well taken - will do.